packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -208,18 +220,21 @@ export default async function processDeployment({
           return deployment;
         }
 
+        if (withLogs) {
+          abortController = displayBuildLogs(client, deployment);
+        }
         output.spinner(
           deployment.readyState === 'QUEUED' ? 'Queued' : 'Building',
           0
         );
       }
 
-      if (event.type === 'building') {
+      if (event.type === 'building' && !withLogs) {
         output.spinner('Building', 0);
```

Comment:
Maybe you can introduce another function that is a little symmetric with `stopSpinner`:

```
function startSpinner(message) {
  if (!withLogs) {
    output.spinner(message, 0)
  }
}
```

And then add a comment next to the 2 functions to explains that for `--logs`, we don't show spinner for reason X.